src/parser/jobs/smn/modules/Weaving.ts

```diff
@@ -1,15 +1,39 @@
 import ACTIONS from 'data/ACTIONS'
+import {CastEvent} from 'fflogs'
+import {dependency} from 'parser/core/Module'
+import CastTime from 'parser/core/modules/CastTime'
 import CoreWeaving, {WeaveInfo} from 'parser/core/modules/Weaving'
 
 const PERMITTED_PHOENIX_WEAVES = 3
 
 export class Weaving extends CoreWeaving {
+	@dependency private castTime!: CastTime
+
 	isBadWeave(weave: WeaveInfo, maxWeaves?: number) {
-		// Permit triple weaves with Firbid Trance because jank
+		// Permit triple weaves with Firebid Trance because Phoenix will not
+		// trigger a Scarlet Flame for the first skill if it is the last weave
+		// in a GCD and cannot always be used as the first weave.
+		// Only a single extra weave will be allowed and only if it is the middle
+		// weave after an instant cast skill.  (If it is the first weave, the triple
+		// is not needed. If it is the third weave, the gain from triple weaving
+		// does not occur.)
 		const hasPhoenixSummon = weave.weaves
 			.some(event => event.ability.guid === ACTIONS.FIREBIRD_TRANCE.id)
-		if (hasPhoenixSummon) {
-			return super.isBadWeave(weave, PERMITTED_PHOENIX_WEAVES)
+		if (hasPhoenixSummon && weave.weaves.length === PERMITTED_PHOENIX_WEAVES) {
+			// Need to make sure we don't allow a triple weave after a hardcast
+			if (!maxWeaves) {
```

Comment:
this block of code here smells like core needs some tweaks to expose its own maxWeaves logic, duping up on that is a bit nasty